### DISCUSSION TITLE
Which UniMath version to ship with Coq Platform 8.17

### DISCUSSION THREAD

**Discussion by benediktahrens** [UniMath/UniMath#1632] [General]
Dear @DanGrayson @m-lindgren @peterlefanulumsdaine @rmatthes @nmvdw ,

The Coq Platform ships a version of UniMath. A new release of Coq is planned for the beginning of March, entailing a new release of the Coq Platform. This gives us an opportunity to update the version of UniMath shipped with the Coq Platform. The maintainers of the Coq Platform have filed UniMath issue https://github.com/UniMath/UniMath/issues/1631 to that end.

The most recent tag on UniMath is from August 2022. Should we make a new tag, of a recent commit, and ask the Coq Platform maintainers to include that version into the next Coq Platform? Are there any considerations that we should take into account when creating the next tag, e.g., should we prefer or avoid a commit before or after a specific date?

Please give your feedback here. Once we have come to a conclusion, we can record that conclusion in https://github.com/UniMath/UniMath/issues/1631 .

Best,
Benedikt

**Comment by m-lindgren** [comment]
Making a new tag sounds good to me. Looking at the things merged since August 2022 I see no reason not to tag the current HEAD of master as a release. If it's possible it would be nice to have #1633 reviewed and merged before creating the release, there's good material in there and we might as well try to include it in the release.

**Comment by peterlefanulumsdaine** [comment]
Agreed with Michael — it’s 6 months since August and there are some good contributions merged, so it feels like there’s good motivation for a new tag.  Is there any particular “to-do list” of things we need to do before fixing a new tagged version — anything beyond what the CI normally checks?

I see from #1631 that they want us to decide “as soon as possible and before March 21st”.  So this is fairly time-sensitive, but not absolutely immediate.

**Comment by rmatthes** [comment]
This week, there will be a discussion involving half of the coordinating committee about the organization of the material on monoidal categories. This could quite much influence my answer (maybe we could do something about that material for that next release, maybe we identify a good point back in history where the need for restructuring it is not yet so apparent).

**Reply by nmvdw** [reply]
I think https://github.com/UniMath/UniMath/pull/1633 can be merged before March 21st. I would say that the main question would be whether we want more changes before creating the tag. Reorganizing the material on monoidal categories is a good candidate for that.

**Comment by rmatthes** [comment]
Yesterday, we not only merged #1633 (that evolved a lot in the reviewing phase also w.r.t. upstream material) but also solved the problems of incompatibility of UniMath and TypeTheory with the dev version of Coq (and even 8.16 for TypeTheory), without letting the other satellite repositories behind. For the next platform release, we should profit from all those achievements. Maybe we could already reveal to the Coq developers that commit that had success of the 12 CI jobs and say that we try to do further reorganisation work by the deadline. So, this commit number would have the role of a fallback.

**Comment by nmvdw** [comment]
I would say that after https://github.com/UniMath/UniMath/pull/1651 is merged (which now should not lead to any problem, see https://github.com/UniMath/UniMath/issues/1614#issuecomment-1462392497), we can make a new tag and ship that version with the Coq platform. Note that this PR now succeeds with Coq dev, so the problem is fixed.

**Comment by benediktahrens** [comment]
I have merged #1651. 

Is there no axiom/admitted statement beyond the desired ones in current master?

**Reply by nmvdw** [reply]
All uses of `Admitted` are commented. I don't think any recent PR introduced an axiom.

**Reply by rmatthes** [reply]
For PR #1646, I did not see admitted statements in the sense of `Admitted` (only in Coq comments). Should there be any?

**Reply by benediktahrens** [reply]
No, on the contrary, there should not be any.

**Comment by rmatthes** [comment]
I still have a concern before shipping: I have the impression that work with ProofGeneral on top of a build with Dune is currently not supported by the configuration file https://github.com/UniMath/UniMath/blob/master/UniMath/.dir-locals.el.in . I work with a system-wide Coq installation, hence delete the line starting with `@LOCAL` (which the makefile would also do). And I add to the end of the list for `coq-prog-args` the following items: 
"-R" "MYPATHTOTHECLONE/UniMath/_build/default/UniMath" "UniMath".
This is the EmacsLisp analogue of the instruction at https://github.com/UniMath/UniMath/blob/master/BUILD_DUNE.md#dune-with-coqtop--proof-general .
This is not a portable solution, and it does not take into account if someone uses `make` or `dune`. If anyone already has a better solution, it would be good to know.

**Comment by rmatthes** [comment]
Okay, making the variable assignment portable is easy:
```
	     (setq coq-prog-args
		   ;; these options should match what's used in ../Makefile
		   `("-quiet" "-emacs" "-noinit" "-indices-matter" "-type-in-type" "-w" "-notation-overridden" "-Q" ,(concat unimath-topdir "UniMath") "UniMath" "-R" ,(concat unimath-topdir "_build/default/UniMath") "UniMath")
		   )
```
I wonder on the choice between `make` and `dune`. Does @DanGrayson have a take on that (as the one who maintained the script in the past)?

**Comment by benediktahrens** [comment]
We have about one more week: "In case you would prefer to see an updated or an older version in the upcoming Coq Platform 2023.03, please inform us as soon as possible and before March 21st, 2023!"

**Comment by benediktahrens** [comment]
Would somebody be willing to make a new tag and, possibly, a new OPAM package for inclusion of UniMath into the Coq Platform?

**Reply by m-lindgren** [reply]
I've never published an opam package before but I can give it a try unless someone else feels inclined. I just have a question regarding the license: 
```
  warning 62: License doesn't adhere to the SPDX standard, see https://spdx.org/licenses/: "Similar to
              MIT license"
```
should this be addressed before release?

**Reply by m-lindgren** [reply]
Testing the current installation with `opam install .` fails for me after 4h with the message `too many arguments` to the install script. Maybe someone with a more poweful computer can test if  they have the same error?

**Comment by rmatthes** [comment]
I never made such a thing, so please do not count on me - I do not even use the platform (I only installed it for testing purposes last year). But I recognize that the deadline for our contribution to the next Coq platform expires tomorrow.

**Comment by benediktahrens** [comment]
I believe that last time the Coq Platform maintainers created the opam package for us, we only had to create the tag. Let's do that tomorrow? The tag should then be indicated in a response to #1631.

**Reply by m-lindgren** [reply]
It is done. There were no PR:s waiting to be merged so I went ahead and created the tag.

**Reply by rmatthes** [reply]
Thank you @m-lindgren. There were definitely PR, but none with mathematical material and instead concerned with the hygiene and maintenance of UniMath.
